pkg/echo/proxy.go

```diff
@@ -13,17 +13,24 @@ import (
 // Proxy is a proxy type for logrus.Logger
 type Proxy struct {
 	dest *slog.Logger
+	ctx  context.Context
 }
 
 var proxy *Proxy
 
-// NewProxyFor creates a new Proxy for a particular logger
-func NewProxyFor(logger *slog.Logger) *Proxy {
+// NewProxyContextFor creates a new Proxy for a particular logger with a particular context.
+func NewProxyContextFor(logger *slog.Logger, ctx context.Context) *Proxy {
 	return &Proxy{
 		dest: logger.With(slog.Bool("echo", true)),
+		ctx:  ctx,
 	}
 }
 
+// NewProxyFor creates a new Proxy for a particular logger
+func NewProxyFor(logger *slog.Logger) *Proxy {
+	return NewProxyContextFor(logger, context.Background())
+}
+
```

Comment:
The actual change does not break the API, the new function `NewProxyContextFor` will be used in CRC to pass in the context on a per-request basis.